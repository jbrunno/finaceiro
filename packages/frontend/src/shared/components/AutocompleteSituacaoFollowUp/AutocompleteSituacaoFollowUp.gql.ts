import { gql } from '@apollo/client';

export const AUTOCOMPLETE_SITUACAO_FOLLOWUP_QUERY = gql`
  query Situacoes($pagination: PaginationInput!, $filter: String) {
    situacoes(pagination: $pagination, filter: $filter) {
      items {
        node {
          id
          codigo
          nome
          descricaoObrigatoria
          textoPadrao
          tempoTrava
        }
      }
      pageInfo {
        totalItems
        hasNextPage
      }
    }
  }
`;

export type PaginationInput = {
  pageNumber: number;
  pageSize: number;
};

export type AutocompleteSituacaoFollowUpQueryVariables = {
  pagination: PaginationInput;
  filter?: string;
};

export type FollowUpSituacao = {
  id: string;
  codigo: string;
  nome: string;
  descricaoObrigatoria: boolean;
  textoPadrao: string;
  tempoTrava: number;
};

export type AutocompleteSituacaoFollowUpData = {
  situacoes: {
    items: Array<{
      node: FollowUpSituacao;
    }>;
    pageInfo: {
      totalItems: number;
      hasNextPage: boolean;
    };
  };
};
