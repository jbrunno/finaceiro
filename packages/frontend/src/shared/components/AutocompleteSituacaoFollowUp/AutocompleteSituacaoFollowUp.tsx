import { TextField, Autocomplete } from '@frontend/components';
import { Clear } from '@frontend/icons';
import { Dispatch, SetStateAction } from 'react';
import { useAutocompleteSituacaoFollowUp } from './hooks/useAutocompleteSituacaoFollowUp';

export type FollowUpSituacao = {
  id: string;
  codigo: number;
  nome: string;
  descricaoObrigatoria: boolean;
  textoPadrao: string;
  tempoTrava: number;
};

export type AutocompleteSituacaoFollowUpProps = {
  label?: string;
  initialValue?: FollowUpSituacao | null;
  situacoes?: FollowUpSituacao[];
  onChange: (value: FollowUpSituacao | null) => void;
  filterSituacoes: Dispatch<SetStateAction<string | undefined>>;
};

export function AutocompleteSituacaoFollowUp({
  onChange,
  initialValue,
  label,
  situacoes,
  filterSituacoes,
}: AutocompleteSituacaoFollowUpProps) {
  const { handleSelectValue, handleClear, filterOptions, value } =
    useAutocompleteSituacaoFollowUp({
      onChange,
      initialValue,
    });

  return (
    <Autocomplete
      autoSelect
      autoComplete
      value={value}
      loading={!situacoes}
      options={situacoes || []}
      loadingText="Carregando..."
      filterOptions={filterOptions}
      clearIcon={<Clear onClick={handleClear} fontSize="small" />}
      noOptionsText="Nada encontrado"
      getOptionLabel={(situacoes) =>
        typeof situacoes === 'string' ? situacoes : situacoes.nome
      }
      isOptionEqualToValue={(option, valueArgs) => option.id === valueArgs.id}
      onChange={(e, value) => {
        if (value && !Array.isArray(value) && typeof value !== 'string') {
          handleSelectValue(value);
        }
      }}
      onInputChange={(_, inputValue) => filterSituacoes(inputValue)}
      renderInput={(params) => (
        <TextField {...params} size="small" label={label} variant="outlined" />
      )}
    />
  );
}
