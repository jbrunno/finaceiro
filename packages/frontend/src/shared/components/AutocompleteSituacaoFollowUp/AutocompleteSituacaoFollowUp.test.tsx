import { render } from '@test-utils/component-testing';
import { AutocompleteSituacaoFollowUp } from './AutocompleteSituacaoFollowUp';

jest.mock('./hooks/useAutocompleteSituacaoFollowUp', () => ({
  useAutocompleteSituacaoFollowUp: () => ({
    label: 'Situação de Follow-up',
    onChange: jest.fn(),
    initialValue: null,
  }),
}));

test('should render AutocompleteSituacaoFollowUp', () => {
  const { container } = render(
    <AutocompleteSituacaoFollowUp
      filterSituacoes={jest.fn}
      onChange={jest.fn}
    />,
  );
  expect(container).toMatchSnapshot();
});
