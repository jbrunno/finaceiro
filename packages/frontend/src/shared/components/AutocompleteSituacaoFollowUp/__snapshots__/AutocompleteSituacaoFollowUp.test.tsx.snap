// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AutocompleteSituacaoFollowUp 1`] = `
.emotion-0.Mui-focused .MuiAutocomplete-clearIndicator {
  visibility: visible;
}

@media (pointer: fine) {
  .emotion-0:hover .MuiAutocomplete-clearIndicator {
    visibility: visible;
  }
}

.emotion-0 .MuiAutocomplete-tag {
  margin: 3px;
  max-width: calc(100% - 6px);
}

.emotion-0 .MuiAutocomplete-inputRoot {
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.MuiAutocomplete-hasPopupIcon.emotion-0 .MuiAutocomplete-inputRoot,
.MuiAutocomplete-hasClearIcon.emotion-0 .MuiAutocomplete-inputRoot {
  padding-right: 30px;
}

.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon.emotion-0 .MuiAutocomplete-inputRoot {
  padding-right: 56px;
}

.emotion-0 .MuiAutocomplete-inputRoot .MuiAutocomplete-input {
  width: 0;
  min-width: 30px;
}

.emotion-0 .MuiInput-root {
  padding-bottom: 1px;
}

.emotion-0 .MuiInput-root .MuiInput-input {
  padding: 4px 4px 4px 0px;
}

.emotion-0 .MuiInput-root.MuiInputBase-sizeSmall .MuiInput-input {
  padding: 2px 4px 3px 0;
}

.emotion-0 .MuiOutlinedInput-root {
  padding: 9px;
}

.MuiAutocomplete-hasPopupIcon.emotion-0 .MuiOutlinedInput-root,
.MuiAutocomplete-hasClearIcon.emotion-0 .MuiOutlinedInput-root {
  padding-right: 39px;
}

.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon.emotion-0 .MuiOutlinedInput-root {
  padding-right: 65px;
}

.emotion-0 .MuiOutlinedInput-root .MuiAutocomplete-input {
  padding: 7.5px 4px 7.5px 5px;
}

.emotion-0 .MuiOutlinedInput-root .MuiAutocomplete-endAdornment {
  right: 9px;
}

.emotion-0 .MuiOutlinedInput-root.MuiInputBase-sizeSmall {
  padding-top: 6px;
  padding-bottom: 6px;
  padding-left: 6px;
}

.emotion-0 .MuiOutlinedInput-root.MuiInputBase-sizeSmall .MuiAutocomplete-input {
  padding: 2.5px 4px 2.5px 8px;
}

.emotion-0 .MuiFilledInput-root {
  padding-top: 19px;
  padding-left: 8px;
}

.MuiAutocomplete-hasPopupIcon.emotion-0 .MuiFilledInput-root,
.MuiAutocomplete-hasClearIcon.emotion-0 .MuiFilledInput-root {
  padding-right: 39px;
}

.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon.emotion-0 .MuiFilledInput-root {
  padding-right: 65px;
}

.emotion-0 .MuiFilledInput-root .MuiFilledInput-input {
  padding: 7px 4px;
}

.emotion-0 .MuiFilledInput-root .MuiAutocomplete-endAdornment {
  right: 9px;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-sizeSmall {
  padding-bottom: 1px;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-sizeSmall .MuiFilledInput-input {
  padding: 2.5px 4px;
}

.emotion-0 .MuiInputBase-hiddenLabel {
  padding-top: 8px;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-hiddenLabel {
  padding-top: 0;
  padding-bottom: 0;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-hiddenLabel .MuiAutocomplete-input {
  padding-top: 16px;
  padding-bottom: 17px;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-hiddenLabel.MuiInputBase-sizeSmall .MuiAutocomplete-input {
  padding-top: 8px;
  padding-bottom: 9px;
}

.emotion-0 .MuiAutocomplete-input {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-overflow: ellipsis;
  opacity: 1;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-2 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  color: #3B3B41DE;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  border-radius: 4px;
  padding-right: 14px;
}

.emotion-2.Mui-disabled {
  color: #3B3B4161;
  cursor: default;
}

.emotion-2:hover .MuiOutlinedInput-notchedOutline {
  border-color: #3B3B41DE;
}

@media (hover: none) {
  .emotion-2:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-2.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #4A31FF;
  border-width: 2px;
}

.emotion-2.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #F44336;
}

.emotion-2.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: #00000042;
}

.emotion-3 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-top: 1px;
  padding: 8.5px 14px;
  padding-right: 0;
}

.emotion-3::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:focus {
  outline: 0;
}

.emotion-3:invalid {
  box-shadow: none;
}

.emotion-3::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-3.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: #3B3B4161;
}

.emotion-3:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-3:-webkit-autofill {
  border-radius: inherit;
}

.emotion-4 {
  position: absolute;
  right: 0;
  top: calc(50% - 14px);
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: #0000008A;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 2px;
  margin-right: -2px;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: transparent;
  }
}

.emotion-5.Mui-disabled {
  background-color: transparent;
  color: #00000042;
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-7 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-8 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-9 {
  float: unset;
  width: auto;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

<div>
  <div
    class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon emotion-0"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-1"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot emotion-2"
      >
        <input
          aria-autocomplete="both"
          aria-expanded="false"
          aria-invalid="false"
          autocapitalize="none"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused emotion-3"
          id="autocomplete-outlined"
          role="combobox"
          spellcheck="false"
          type="text"
          value=""
        />
        <div
          class="MuiAutocomplete-endAdornment emotion-4"
        >
          <button
            aria-label="Open"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator emotion-5"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root emotion-7"
            />
          </button>
        </div>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline emotion-8"
        >
          <legend
            class="emotion-9"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
